lib/src/barback/compiler_mode.dart

```diff
@@ -0,0 +1,5 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+enum CompilerMode { Dart2Js, DevCompiler, None }
```

Comment:
[nit] is there precedence within this package for capitalized enum values? Typically we use lower camel case